platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -0,0 +1,309 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hillview.storage;
+
+import org.hillview.table.api.*;
+import org.apache.cassandra.config.CFMetaData;
+import org.apache.cassandra.config.ColumnDefinition;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.cql3.ColumnIdentifier;
+import org.apache.cassandra.db.rows.Cell;
+import org.apache.cassandra.db.rows.ColumnData;
+import org.apache.cassandra.db.rows.ComplexColumnData;
+import org.apache.cassandra.db.rows.Row;
+import org.apache.cassandra.db.rows.Unfiltered;
+import org.apache.cassandra.db.rows.UnfilteredRowIterator;
+import org.apache.cassandra.io.sstable.Descriptor;
+import org.apache.cassandra.io.sstable.ISSTableScanner;
+import org.apache.cassandra.io.sstable.format.SSTableReader;
+import org.hillview.table.ColumnDescription;
+import org.hillview.table.Schema;
+import org.hillview.table.Table;
+import org.hillview.table.membership.FullMembershipSet;
+import org.hillview.utils.HillviewLogger;
+
+import javax.annotation.Nullable;
+import java.util.stream.Stream;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.StreamSupport;
+
+import org.apache.cassandra.db.PartitionColumns;
+import org.apache.cassandra.db.SerializationHeader;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.UTF8Type;
+import org.apache.cassandra.dht.*;
+import org.apache.cassandra.io.sstable.metadata.MetadataComponent;
+import org.apache.cassandra.io.sstable.metadata.MetadataType;
+import org.apache.cassandra.utils.FBUtilities;
+
+/**
+ * Knows how to read a local Cassandra's SSTable file.
+ */
+public class CassandraSSTableLoader {
+    @Nullable
+    private Schema actualSchema;
+    @Nullable
+    private final String ssTablePath;
+    @Nullable
+    IAppendableColumn[] columns;
+    @Nullable
+    private String currentToken;
+
+    private int currentColumn;
+    private int currentRow;
+    private boolean allowFewerColumns;
+
+    static
+    {
+        /** Initializing DatabaseDescriptor that contains Cassandra's general configuration */
+        DatabaseDescriptor.clientInitialization(false);
+    }
+    
+    public CassandraSSTableLoader(String ssTablePath) {
+        this.ssTablePath = ssTablePath;
+    }
+
+    public CFMetaData getSSTableMetadata(Descriptor desc) throws IOException
+    {
+        if (!desc.version.storeRows())
+            throw new IOException("pre-3.0 SSTable is not supported.");
```

Comment:
perhaps you can throw just RuntimeException?

Reply:
Fixed it.